### PR TITLE
Bugfixes

### LINKED ISSUE #103: Creating an account on Firefox does not work.
User hits "create" and the dialogue goes away but nothing appears to happen and the account is not created. This issue seems to happen only on Firefox, not Chrome.

### PR DESCRIPTION
**Pull request by macfarlandian**
commits ce6be3924dd14f4965746167e315b150af3745dc and ab38064d774add82da9c33f3b7e00d232cec5331 fix #103 (corrects a bug in the login process affecting firefox users)

### CODE REVIEW

**Comment by macfarlandian** [comment]
@salspaugh don't merge this yet, i'm going to push more bugfixes

**Comment by salspaugh** [comment]
Ok! Let me know when I should review it?

**Comment by macfarlandian** [comment]
@salspaugh what else do you see as a high-priority fix? i looked into #104 but haven't been able to replicate it so far. should we triage? or have you done some of that already?

**Comment by salspaugh** [comment]
I went through the list and noticed that one high-priority one wasn't there, so I added it (#107).

I've tried to do some categorization of the rest. What do you think of these? Feel free to suggest alternative assignments of issues to categories.

**FIXED** and **CAN'T REPRODUCE** are done or should be ignored. 

**SHOULD FIX** are things that are bugs that we should try to get rid of for sure, and **NICE TO HAVE** are bugs that would be nice to fix if we have time. I think we should do the **SHOULD FIX** first and foremost, then make the website all functional and presentable. If time permits we could do the **NICE TO HAVE**, then further prioritize the remaining ones.

**AESTHETIC CHANGES** are lowest priority and **WONT FIX** we should just ignore I think.

**FIXED**
* 103

**CAN’T REPRODUCE**
* 104

**SHOULD FIX**
* 107
* 23 (optionally 7)
* 93

**NICE TO HAVE**
* 24
* 58
* 101

**AESTHETIC CHANGES**
* 73
* 67
* 78
* 77
* 76
* 75
* 74

**WONTFIX**
* 80
* 79

**Comment by macfarlandian** [comment]
sounds good to me ... i will try to get the rest of the SHOULD FIX bugs done in this PR

**Comment by salspaugh** [comment]
Ok! For #23 / #7 (I think only doing one of these is fine and #23 would be easier), it probably makes sense just to impose an arbitrary file size limit, so that we can slow the rate at which our database gets out of control.

Speaking of which, it probably makes sense to copy the current data out of the database and archive it somewhere, then start with a fresh one once we "go public" with the site. Thoughts?

**Comment by salspaugh** [comment]
Another thing we can do that might be good is to create an example user with pre-uploaded datasets for the people who just want to poke around to see what it's about versus actually use it for anything.

**Comment by macfarlandian** [comment]
@salspaugh I've been working on some error-handling code to address #23 but I think a demo user account is a good idea also. 

one option would be to pre-load some data to an account and disable the upload/delete functions so people can't screw it up too much, since it will be shared by the public.

another would be to spin out temporary demo accounts for people who want to check it out and then have them self-destruct after a few hours. that way you don't have weird conflicts happening if multiple people are logged into the same demo account.

any preference? or other suggestions?

**Comment by salspaugh** [comment]
I liked the upload / delete function idea best at first but then you brought up the issue of weird things happening through concurrent use, which is a good point. How hard would it be to do the temporary account thing?

**Comment by macfarlandian** [comment]
not too hard i don't think ... what's a reasonable time limit before
self-destruction?

**Comment by salspaugh** [comment]
Maybe like an hour? 

It would be good to put a notification like "This is a temporary account that will be deleted in one hour. Please email <I guess my email> to request a permanent account."

**Comment by macfarlandian** [comment]
ok i'll add that to my list

**Comment by macfarlandian** [comment]
@salspaugh this PR is ready for your review & merge, all of the "should fix" bugs are addressed

File: client/charts-directive.js [hunk 1]
```diff
@@ -50,21 +50,7 @@ var renderStringChart = function(scope, dimensions) {
      .attr("y", function(d, i) { return yScale(i); })
      .attr("width", function(d) { return xScale(d.freq); })
      .attr("height", yScale.rangeBand())
-     .on("mouseover", function(d){
-       var tooltip = d3.select("body")
```
Comment (salspaugh) [inline client/charts-directive.js]: For some reason at the moment I have a change pending to this line in my local version that changes it to:

`var tooltip = d3.select(divID)`

I don't know why I made this change. :( It's been a while. 

Just saying ... for posterity.

**Comment by salspaugh** [comment]
@macfarlandian How do I get to the actual dataframer tool instead of the landing page? Sorry for the dumb questions -- am overwhelmed with various work!

**Comment by macfarlandian** [comment]
ha sorry it's at /demo

**Comment by salspaugh** [comment]
When you click on the DataFramer "logo" in the tool, it no longer re-directs to the datasets page, but to the landing page. Should we change that? Otherwise people can't go back to the list of datasets as far as I know without changing the URL directly.

**Comment by macfarlandian** [comment]
that's addressed in the other PR with the landing page content

**Comment by salspaugh** [comment]
Ok sounds good. 

I checked the other bugs and everything looks good. The only small comment I have is that I had to dig around my filesystem for a .csv file bigger than 10MB so that is plenty high and could be reduced if it we want.

**Comment by salspaugh** [comment]
Oh so in summary :+1: merge it!

**Pull request merged**